src/main/java/com/adq/jenkins/xmljobtodsl/InitialArgumentsHandler.java

```diff
@@ -133,36 +140,56 @@ public static String getJobNameBasedOnPath(File file) {
 	}
 
 	public static String getJobPromotedBuildsXMLs(File file)
-		throws IOException {
+			throws IOException, SAXException, ParserConfigurationException {
 
 		IOUtils ioUtils = new IOUtils();
+		ArrayList<String> promotionNames = new ArrayList<>();
+
+		// parse xmlFile to get the names of promotion directories and add to list
+		DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+		DocumentBuilder db =  dbf.newDocumentBuilder();
+		Document doc = db.parse(file);
+
+		NodeList promotionList = doc.getDocumentElement().getElementsByTagName("activeProcessNames");
+
+		for (int i = 0; i < promotionList.getLength(); ++i) {
```

Comment:
you could use a normal for loop here, no?

```for(promotion : promotionList){}```

Reply:
We already discussed but linking this here: Since it's a nodelist I can't iterate using forEach and can only access the items by their index number. https://docs.oracle.com/javase/7/docs/api/org/w3c/dom/NodeList.html.